install/kubernetes/Makefile

```
@@ -29,19 +29,30 @@ HELM_DOCS := $(DOCKER_RUN) $(HELM_DOCS_IMAGE)
 LOGO_BASE_URL := https://cdn.jsdelivr.net/gh/cilium
 LOGO_PATH := Documentation/images/logo-solo.svg
 
-all: update-versions docs lint
+define generate_values_yaml
+echo -e "# File generated by $(RELATIVE_DIR)/Makefile; DO NOT EDIT." > $(2); \
+echo -e "# This file is based on $(RELATIVE_DIR)/cilium/values.yaml.tmpl.\n" >> $(2); \
+envsubst < $(1) >> $(2)
+endef
+
+all: update-versions check-values-yaml docs lint
 
 update-chart: $(ROOT_DIR)/VERSION # Update chart versions to point to the current version.
 	$(ECHO_GEN)cilium/Chart.yaml
 	$(QUIET)grep -lR -e 'version:' -e 'appVersion:' $(CILIUM_CHARTS)/ \
 		| xargs -L 1 $(SED) -i -e 's/'$(CILIUM_CHART_REGEX)'/\1 '$(VERSION)'/g'
 	$(QUIET)$(SED) -i 's;icon:.*;icon: $(LOGO_BASE_URL)/cilium@$(CILIUM_BRANCH)/$(LOGO_PATH);' $(CHART_FILE)
 
+check-values-yaml:
+	$(ECHO_CHECK) $@
+	$(QUIET)( $(call generate_values_yaml,$(CILIUM_CHARTS)/values.yaml.tmpl,/dev/stdout) ) \
+		| diff - "$(CILIUM_CHARTS)/values.yaml" \
```

nit: how about `-u` so that the output is unified diff format?
```suggestion
		| diff -u - "$(CILIUM_CHARTS)/values.yaml" \
```

Note that if this suggestion is accepted, the commit message will need to be updated accordingly (i.e. the diff output will differ).

---

I considered that, but `-u` is not a [standard flag](https://pubs.opengroup.org/onlinepubs/009604499/utilities/diff.html). If all common `diff` utilities support it, then I'm fine to do it, but I lack the knowledge of what e.g. macOS ships by default.

---

According to https://ss64.com/osx/diff.html it seems supported. Let's try it. Easy to revert if it breaks any exotic setup.

---

@gandro FWIW your link is _POSIX.1-2001_, updated by _POSIX.1-2008_ which [defines -u for diff](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/diff.html).

---

Oh neat! :partying_face: Thanks for the hint